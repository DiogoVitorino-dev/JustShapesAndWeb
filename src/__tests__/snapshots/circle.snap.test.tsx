import { render, renderHook } from "@testing-library/react-native";
import { useSharedValue } from "react-native-reanimated";

import Circle, {
  CirclePosition,
  CircleRadius,
} from "@/models/geometric/circle";

describe("Circle model - Snapshot test", () => {
  let pos: CirclePosition;
  let radius: CircleRadius;

  it("Should renders correctly", () => {
    renderHook(() => {
      pos = useSharedValue({ x: 25, y: 50 });
      radius = useSharedValue(90);
    });

    const tree = render(<Circle radius={radius} position={pos} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
