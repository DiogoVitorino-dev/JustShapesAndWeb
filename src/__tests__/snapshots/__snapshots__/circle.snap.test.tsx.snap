// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle model - Snapshot test Should renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "borderRadius": 45,
        "height": 90,
        "left": 25,
        "top": 50,
        "width": 90,
      },
    }
  }
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(255, 64, 20)",
      "borderRadius": 45,
      "height": 90,
      "left": 25,
      "position": "absolute",
      "top": 50,
      "width": 90,
    }
  }
  testID="circleModel"
/>
`;
